bundles/remote_services/rsa_common/src/remote_interceptors_handler.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include <stdlib.h>
+
+#include "celix_bundle_context.h"
+#include "celix_constants.h"
+#include "utils.h"
+
+#include "remote_interceptors_handler.h"
+
+typedef struct entry {
+    const celix_properties_t *properties;
+    remote_interceptor_t *interceptor;
+} entry_t;
+
+struct remote_interceptors_handler {
+    celix_array_list_t *interceptors;
```

Comment:
protect this with a celix mutex

Reply:
is the protection also missing for the pubsub interceptor handler?

Reply:
You are right, will add it. To both the psa and rsa handler